index.js

```diff
@@ -186,6 +191,16 @@ Router.prototype.handle = function handle(req, res, callback) {
   req.baseUrl = parentUrl
   req.originalUrl = req.originalUrl || req.url
 
+  // trigger the "beginning of route handling" event
+  if (events && 'handlestart' in events) self.emit('handlestart', req)
+
+  // trigger the "end of route handling" event
+  if (events && 'handleend' in events) {
+    res.once('finish', function () {
```

Comment:
What is the purpose of adding the finish event listener? Especially because there are no tests around all the edge cases this will cause (for example, are we saying that if the response is already finished before the handleend, our events is not supposed to fire?)